### ISSUE TITLE
Going forward with Cargo release

### ISSUE THREAD

**Issue by willeagren** [Ignalina/alloy#40]
What features/functionality do we require for a mininmal release?, what do we want to include in future releases?, how do we define a idiomatic API in both Go + Rust?, ... etc

**Comment by willeagren** [comment]
Cargo styled crates are usually only APIs, not entire solutions. So I think we should brainstorm a number of features from the Rust side that we want. We should not, or even can't, publish alloy in its current state.

**Comment by Ignalina** [comment]
you are right , .. alloy fits as an go module .. only need to "add " the static build rust impl .. and it will follow along the Go modul. keeping the built rust impl thin , with runtime pluggable rust implementation (the code that acutally does something with the arrow data..)

Im drawing a alloy picture with above including idea
It just struck me ... ffi/impl.h is resuable from multiple target

**Comment by willeagren** [comment]
Can you include pre-compiled binaries in Go modules? In that case, yea, we should just build the Rust code and include it then.

**Comment by willeagren** [comment]
But releasing the Rust code on Cargo sounds like fun... haha

**Comment by Ignalina** [comment]
You got wok to do , ignalina/thund.rs created and you added , do create a delta.rs utilizing CDYLIB as a first try.
We then make a cargo relase of of thund.rs and use Thund's upcoming alloy step as a driver :)   We wrap this up in a thund-dist container with a demo DAG
{readcsv_into_arrow, duckdbProcess,Aloy->deltalake (  Wilhems thund.rs delta step )



A picture of this , we need to have :)

and yes i included lib.rs in git , to easy test

**Comment by Ignalina** [comment]
Ok thund depend o alloy now, and can be used to test Alloy as an module

**Comment by willeagren** [comment]
Wow snyggt!

Tittade på dina ändringar i master, coola grejer. Så en kommentar; " // Yes
Horrificly using overisized constant array until we can transfer a dynamic
sized slice/list via FFI ", men jag har på feature/rust skrivit kåd för
dyamic slice som byggs upp, potentiellt, file IO eller dylikt. Kan titta på
o lägga in det i master?

Den mån 17 okt. 2022 kl 02:15 skrev Rickard Ernst Björn Lundin <
***@***.***>:

**Comment by Ignalina** [comment]
cool , go ahead with the kåd for dynamic slice if you havent merged it yet :)

**Comment by willeagren** [comment]
It is merged on main

**Comment by Ignalina** [comment]
Thanks !  I had problems build thund .. but discovered I had to to "git lfs install" withint the thund clone directory ....
its totally ok ..hmm i wonder how Apache Arrow project have solved it ..they also need .a libraries

**Comment by Ignalina** [comment]
btw the slice thingy is really nice you now are a Go programmer ! , just saw the readme output also !

Try clone and build this one , it should break if you dont do "git lfs install"
https://github.com/Ignalina/thund
